scripts/dev/dump_diagnostic.py

```diff
@@ -0,0 +1,120 @@
+    the result as a new dictionary or list.
+    """
+    if isinstance(value, list):
+        return [clean_nones(x) for x in value if x is not None]
+    if isinstance(value, dict):
+        return {key: clean_nones(val) for key, val in value.items() if val is not None}
+    return value
+
+
+def header(msg: str) -> str:
+    dashes = (
+        "----------------------------------------------------------------------------"
+    )
+    return "\n{0}\n{1}\n{0}\n".format(dashes, msg)
+
+
+def dump_crd(crd_log: typing.TextIO):
+    crdv1 = client.ApiextensionsV1beta1Api()
+    try:
+        crd_log.write(header("CRD"))
+        mdb = crdv1.list_custom_resource_definition(pretty="true")
+        crd_log.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_custom_resource_definition: %s\n" % e)
+
+
+def dump_persistent_volume(diagnosticFile: typing.TextIO):
+    corev1 = client.CoreV1Api()
+    try:
+        diagnosticFile.write(header("Persistent Volume Claims"))
+        mdb = corev1.list_persistent_volume(pretty="true")
+        diagnosticFile.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_persistent_volume %s\n" % e)
+
+
+def dump_stateful_sets_namespaced(diagnosticFile: typing.TextIO, namespace: str):
+    av1beta1 = client.AppsV1Api()
+    try:
+        diagnosticFile.write(header("Stateful Sets"))
+        mdb = av1beta1.list_namespaced_stateful_set(namespace, pretty="true")
+        diagnosticFile.write(yaml.dump(clean_nones(mdb.to_dict())))
+    except ApiException as e:
+        print("Exception when calling list_namespaced_stateful_set: %s\n" % e)
+
+
+def dump_pod_log_namespaced(namespace: str, name: str):
+    corev1 = client.CoreV1Api()
+    try:
+        if name.startswith("mdb0"):
+            pod_file_mongoDB_agent = open(
+                "logs/e2e/{}-mongodb-agent.log".format(name), "w"
+            )
+            pod_file_mongod = open("logs/e2e/{}-mongod.log".format(name), "w")
+            log_mongodb_agent = corev1.read_namespaced_pod_log(
+                name=name, namespace=namespace, pretty="true", container="mongodb-agent"
+            )
+            log_mongod = corev1.read_namespaced_pod_log(
+                name=name, namespace=namespace, pretty="true", container="mongod"
+            )
+            pod_file_mongoDB_agent.write(log_mongodb_agent)
+            pod_file_mongod.write(log_mongod)
+            pod_file_mongod.close()
+            pod_file_mongoDB_agent.close()
+        elif name.startswith("mongodb-kubernetes-operator"):
+            pod_file = open("logs/e2e/{}.log".format(name), "w")
+            log = corev1.read_namespaced_pod_log(
+                name=name, namespace=namespace, pretty="true"
+            )
+            pod_file.write(log)
+            pod_file.close()
+    except ApiException as e:
```

Comment:
The scope of this exception is very broad, it's not clear which operation can throw an `ApiException`